src/containers/Dashboard/saga.js

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import { all, call, takeLatest, put } from 'redux-saga/effects';
+
+import appConstants from './constants';
+import { updateRandomBannerProducts, setLoading, setError } from './actions';
+import { showErrorToast } from '../../utils/common';
+
+function* fetchBannerAsync(action) {
+  try {
+    const { cntProducts } = action;
+
+    yield put(setLoading(true));
+
+    const request = () =>
+      axios.get('http://www.teamcofounder.com/api/v1/mednet/bannerProducts');
+    const response = yield call(request);
+
+    const shuffled = response.data.sort(() => 0.5 - Math.random());
+    let randomItems = shuffled.slice(0, cntProducts);
+
+    yield put(updateRandomBannerProducts(randomItems));
+
+    yield put(setLoading(false));
+  } catch (error) {
+    yield put(setLoading(false));
+    yield put(setError(true));
```

Comment:
you can set error as error text instead of just boolean. Then you can use that error text in component, not only as boolean flag.